packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -490,10 +504,20 @@ export default createRule<Options, MessageId>({
         );
 
         if (isVoidReturningFunctionType(checker, tsNode.name, contextualType)) {
-          context.report({
-            node: node.value,
-            messageId: 'voidReturnProperty',
-          });
+          const functionNode = findFunctionNode(node.value);
+
+          if (functionNode.returnType) {
+            context.report({
+              node: functionNode.returnType,
```

Comment:
Looks like the actual `typeAnnotation` is what we want; seemingly the `returnType` AST node includes the `: T`, not just `T` 🤷 
```suggestion
              node: functionNode.returnType.typeAnnotation,
```

WDYT?